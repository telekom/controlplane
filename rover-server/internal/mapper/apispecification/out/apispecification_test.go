// Copyright 2025 Deutsche Telekom IT GmbH
//
// SPDX-License-Identifier: Apache-2.0

package out

import (
	"github.com/gkampitakis/go-snaps/match"
	"github.com/gkampitakis/go-snaps/snaps"
	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"
)

var _ = Describe("ApiSpecificationResponse Mapper", func() {
	Context("MapRequest", func() {
		It("must map a ApiSpecification to an ApiSpecificationResponse correctly", func() {
			output, err := MapResponse(apiSpecification)

			Expect(err).To(BeNil())

			Expect(output).ToNot(BeNil())
			snaps.MatchJSON(GinkgoT(), output, match.Any("status.time"))
		})

		It("must return an error if the input ApiSpecification is nil", func() {
			output, err := MapResponse(nil)

			Expect(output).ToNot(BeNil())
			snaps.MatchSnapshot(GinkgoT(), output)

			Expect(err).ToNot(BeNil())
			Expect(err.Error()).To(ContainSubstring("input api specification is nil"))
		})

	})
})
