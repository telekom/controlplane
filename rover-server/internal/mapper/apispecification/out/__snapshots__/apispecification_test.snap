
[ApiSpecificationResponse Mapper MapRequest must map a ApiSpecification to an ApiSpecificationResponse correctly - 1]
{
 "category": "other",
 "id": "eni--hyperion--apispec-sample",
 "name": "",
 "specification": {
  "info": {
   "title": "Rover API",
   "version": "1.0.0"
  },
  "openapi": "3.0.0",
  "servers": {
   "url": "http://rover-api/eni/distr/v1"
  }
 },
 "status": {
  "processingState": "done",
  "state": "complete",
  "time": "<Any value>"
 },
 "vendorApi": false
}
---

[ApiSpecificationResponse Mapper MapRequest must return an error if the input ApiSpecification is nil - 1]
api.ApiSpecificationResponse{}
---
